src/rpc/rawtransaction.cpp

```diff
@@ -287,7 +283,7 @@ static RPCHelpMan gettxoutproof()
     LOCK(cs_main);
 
     if (pblockindex == nullptr) {
-        const CTransactionRef tx = GetTransaction(/* block_index */ nullptr, /* mempool */ nullptr, oneTxid, Params().GetConsensus(), hashBlock);
+        const CTransactionRef tx = GetTransaction(/* block_index */ nullptr, /* mempool */ nullptr, *setTxids.begin(), Params().GetConsensus(), hashBlock);
```

Comment:
in commit 73dc19a330f8cb063af46e6c4246f2e64a04bdc1:

Just for future reference, this is reading uninitialized memory

Reply:
Care to explain?

Reply:
std::set::begin will return the end interator if the set is empty.

Dereferencing the end pointer is accessing uninitialized memory.

Reply:
Right, would you check `!setTxids.empty()` again?

Reply:
the bug only exists in commit 73dc19a , which can't be modified because it is already merged

Reply:
Thanks for spotting @MarcoFalke.
I guess this needs for a follow up PR.

Reply:
The bug is fixed in the 2nd commit 52fc39917fc52c2ff279fe434431e18900b347bd.

Reply:
Now I got it. An intermediate commit had this uninitialized memory access. No need for action.